### DISCUSSION TITLE
Exposing Fallthrough attributes on consumer component

### DISCUSSION THREAD

**Discussion by trc-mathieu** [vuejs/language-tools#3361] [Ideas]
Fallthrough attributes are nice for wrapping components: https://vuejs.org/guide/components/attrs.html

But is there a way to automatically expose what are the props, emits, slots that can be used from intellisense point of view?

Can the metadata be automatically extracted from the wrapped component?
Not sure if it is related to https://github.com/vuejs/language-tools/issues/3201

**Comment by rchl** [comment]
I can't speak for whether VLT can extract the state of `inheritAttrs` and use that to determine whether to allow fall-through attributes or not but... consider the default case where fall-through attributes are allowed... While I think that VLT should allow those as Vue allows them, this now creates an issue that any unknown prop/attribute is allowed on a custom component. That means that it's no longer possible to catch issues like when the prop was misnamed or when a prop within custom component was renamed and users of the component did not get updated to align with that. So I think that conceptually it's a difficult problem to solve as both ways seem to have disadvantages.

Unless I'm missing something and there exists a good hypothetical solution that would satisfy both use-cases...

**Reply by rchl** [reply]
And BTW, `class` and `style` should already be allowed. At least they are in Vue 3.x and for Vue 2.x there is a PR that I've created.

So those common cases are covered (although probably don't take `inheritAttrs` into account.

So the problem is with `id` and any other random attribute.

**Reply by trc-mathieu** [reply]
Maybe this should be a vue feature then. A new macro maybe that exposes the same props/emits/slots as the wrapped components? Otherwise, we completely loose the DX and wrapping components is like going back before typescript and all these wonderful tools like volar existed.

i.e.
```
<script setup>
  import {VBtn} from "vuetify/components"
  defineWrapComponent<VBtn>()
</script>
<template>
  <VBtn>Wraping a vuetify button</VBtn>
</template>

```
